doc/Type/Buf.rakudoc

```diff
@@ -120,72 +118,79 @@ Returns a C<List> of integers.
+Adds elements at the end of the buffer.
 
-     my @φ =  1,1, * + * … ∞;
-     my $bú = Buf.new( @φ[^5] );
-     $bú.push( @φ[5] );
+     my $bú = Buf.new( 1, 1, 2, 3, 5 );
+     $bú.push( 8 );
      say $bú.raku; # OUTPUT: «Buf.new(1,1,2,3,5,8)␤»
 
 =head2 method pop
 
     method pop()
 
-Extracts the last element of the buffer
+Returns and removes the last element of the buffer.
 
-=for code :preamble<my $bú>
-say $bú.pop(); # OUTPUT: «8␤»
-say $bú.raku; # OUTPUT: «Buf.new(1,1,2,3,5)␤»
+     my $bú = Buf.new( 1, 1, 2, 3, 5 );
+     say $bú.pop(); # OUTPUT: «5␤»
+     say $bú.raku;  # OUTPUT: «Buf.new(1,1,2,3)␤»
 
 =head2 method append
 
     method append( $elems )
 
-Appends at the end of the buffer
+Appends to the end of the buffer.
 
-=for code :preamble<my $bú; my @φ>
-$bú.append( @φ[5..10] );
-say $bú.raku; # OUTPUT: «Buf.new(1,1,2,3,5,8,13,21,34,55,89)␤»
+     my $bú = Buf.new( 1, 1, 2, 3, 5 );
+     $bú.append(9, 8, 7, 6);
+     say $bú.raku; # OUTPUT: «Buf.new(1,1,2,3,5,9,8,7,6)␤»
 
 =head2 method prepend
 
     method prepend( $elems )
 
-Adds elements at the beginning of the buffer
+Inserts elements at the beginning of the buffer.
+
+    my $bú = Buf.new( 1, 1, 2, 3, 5 );
+    $bú.prepend( 0 );
+    say $bú.raku; # OUTPUT: «Buf.new(0,1,1,2,3,5)␤»
+
+=for comment
+Duplicated from Array.prepend
 
-=for code :preamble<my $bú>
-$bú.prepend( 0 );
-say $bú.raku; # OUTPUT: «Buf.new(0,1,1,2,3,5,8,13,21,34,55,89)␤»
+The difference from method C<unshift> is that if you prepend a B<single> array
+or list argument, C<prepend> will flatten that array / list, whereas C<unshift>
+prepends the list / array as just a single element.
 
 =head2 method shift
 
     method shift()
 
-Takes out the first element of the buffer
+Removes and returns the first element of the buffer.
 
-=for code :preamble<my $bú>
-$bú.shift();
-say $bú.raku; # OUTPUT: «Buf.new(1,1,2,3,5,8,13,21,34,55,89)␤»
+    my $bú = Buf.new( 1, 1, 2, 3, 5 );
+    say $bú.shift(); # OUTPUT: «1␤»
+    say $bú.raku;    # OUTPUT: «Buf.new(1,2,3,5)␤»
 
 =head2 method unshift
 
     method unshift()
 
-Adds elements at the beginning of the buffer
+Inserts elements at the beginning of the buffer.
 
-=for code :preamble<my $bú>
-$bú.unshift( 0 );
-say $bú.raku; # OUTPUT: «Buf.new(0,1,1,2,3,5,8,13,21,34,55,89)␤»
+    my $bú = Buf.new( 1, 1, 2, 3, 5 );
+    $bú.unshift( 0 );
+    say $bú.raku; # OUTPUT: «Buf.new(0,1,1,2,3,5)␤»
 
 =head2 method splice
 
     method splice( Buf:D: $start = 0, $elems?, *@replacement --> Buf)
 
-Substitutes elements of the buffer by other elements
+Substitutes elements of the buffer by other elements, returning a buffer
+with the removed elements.
 
-=for code :preamble<my $bú>
-$bú.splice:  0, 3, <3 2 1>;
-say $bú.raku; # OUTPUT: «Buf.new(3,2,1,2,3,5,8,13,21,34,55,89)␤»
+    my $bú = Buf.new( 1, 1, 2, 3, 5 );
+    say $bú.splice:  0, 3, <3 2 1>;  # OUTPUT: «Buf:0x<01 01 02>␤»
+    say $bú.raku;                    # OUTPUT: «Buf.new(3,2,1,2,3,5)␤»
```

Comment:
Seems to me that this output is actually wrong, and it was wrong previously as well. It will be `3, 2, 1, 3, 5`, without the two.

Reply:
good catch, fixed.